utils_nlp/models/transformers/abstractive_summarization_bertsum.py

```diff
@@ -0,0 +1,815 @@
+            batch_size=batch_size,
+            collate_fn=collate_fn,
+        )
+        print("dataset length is {}".format(len(test_dataset)))
+
+        def format_summary(translation):
+            """ Transforms the output of the `from_batch` function
+            into nicely formatted summaries.
+            """
+            raw_summary = translation
+            summary = (
+                raw_summary.replace("[unused0]", "")
+                .replace("[unused3]", "")
+                .replace("[CLS]", "")
+                .replace("[SEP]", "")
+                .replace("[PAD]", "")
+                .replace("[unused1]", "")
+                .replace(r" +", " ")
+                .replace(" [unused2] ", ".")
+                .replace("[unused2]", "")
+                .strip()
+            )
+
+            return summary
+
+        def generate_summary_from_tokenid(preds, pred_score):
+            batch_size = preds.size()[0]  # batch.batch_size
+            translations = []
+            for b in range(batch_size):
+                if len(preds[b]) < 1:
+                    pred_sents = ""
+                else:
+                    pred_sents = self.processor.tokenizer.convert_ids_to_tokens(
+                        [int(n) for n in preds[b] if int(n) != 0]
+                    )
+                    pred_sents = " ".join(pred_sents).replace(" ##", "")
+                translations.append(pred_sents)
+            return translations
+
+        generated_summaries = []
+        from tqdm import tqdm
+
+        for batch in tqdm(
+            test_dataloader, desc="Generating summary", disable=not verbose
+        ):
+            input = self.processor.get_inputs(batch, device, "bert", train_mode=False)
+            translations, scores = predictor(**input)
+
+            translations_text = generate_summary_from_tokenid(translations, scores)
+            summaries = [format_summary(t) for t in translations_text]
+            generated_summaries.extend(summaries)
+        return generated_summaries
+
+    def save_model(self, global_step=None, full_name=None):
+        """
+        save the trained model.
+
+        Args:
+            global_step (int, optional): The number of steps that the model has been finetuned for. Defaults to None.
+            full_name (str, optional): File name to save the model's `state_dict()`. If it's None,
+                the model is going to be saved under "fine_tuned" folder of the cached directory
+                of the object. Defaults to None.
+        """
+        model_to_save = (
+            self.model.module if hasattr(self.model, "module") else self.model
+        )  # Take care of distributed/parallel training
+
+        if full_name is None:
+            output_model_dir = os.path.join(self.cache_dir, "fine_tuned")
+            os.makedirs(self.cache_dir, exist_ok=True)
+            os.makedirs(output_model_dir, exist_ok=True)
+            full_name = os.path.join(output_model_dir, name)
```

Comment:
name doesn't seem to be defined anywhere

Reply:
right. will need to provide a default name.